### PR TITLE
Adding the ability to do row selection within the table component.

### PR DESCRIPTION
**Pull request by Shadowblade911**
1) This won't solve any issues involving remaking the state of the table. That's beyond the scope of this story.
2) styling should be handled by the consumer, not the table.

3) ... I really dun goofed with webpack. I fixed some issues I found in here as well.

### CODE REVIEW

File: src/centriamTable/CentriamTable.js [hunk 1]
```diff
@@ -47,11 +49,22 @@ export default class CentriamTable extends React.Component {
...
-            rowClick: props.rowClick  || noop, // noop function
             sortInfo: props.sortInfo || new SortInfo(),
-
+            selectedRow: props.selectedRow
         };
 
+        this.rowClick = rowSelection ? function(data){
+            this.setState(Object.assign({}, this.state, {selectedRow : data}));
+            rowClick(data);
+        }.bind(this) : rowClick;
```
Comment (karlrwjohnson) [inline src/centriamTable/CentriamTable.js]: Lambdas don't make you do this. :)

`(data) => { this.setState() }`

File: src/centriamTable/CentriamTable.js [hunk 2]
```diff
@@ -47,11 +49,22 @@ export default class CentriamTable extends React.Component {
...
             data: props.data,
             rowHeight: props.rowHeight,
-            rowClick: props.rowClick  || noop, // noop function
             sortInfo: props.sortInfo || new SortInfo(),
-
+            selectedRow: props.selectedRow
         };
 
+        this.rowClick = rowSelection ? function(data){
+            this.setState(Object.assign({}, this.state, {selectedRow : data}));
```
Comment (karlrwjohnson) [inline src/centriamTable/CentriamTable.js]: I thought setState took care of merging these for you

Comment (Shadowblade911) [reply src/centriamTable/CentriamTable.js]: Yes/No. It's supposed to, but I've had problems in the past when merging large objects together. This is just an extremely explicit way of doing it and results in fewer bugs

File: src/centriamTable/CentriamTable.js [hunk 3]
```diff
@@ -47,11 +49,22 @@ export default class CentriamTable extends React.Component {
         self.state = {
             data: props.data,
             rowHeight: props.rowHeight,
-            rowClick: props.rowClick  || noop, // noop function
             sortInfo: props.sortInfo || new SortInfo(),
-
+            selectedRow: props.selectedRow
         };
 
+        this.rowClick = rowSelection ? function(data){
```
Comment (karlrwjohnson) [inline src/centriamTable/CentriamTable.js:56]: Idea: If you're already keeping track of which row's selected, then you can broadcast another event when the row selection changes. If a row is already selected, you can detect it and not broadcast it.

**Review by karlrwjohnson — APPROVED** [review]

File: src/centriamTable/CentriamTable.js (hunk 2 above)
Comment (karlrwjohnson) [reply src/centriamTable/CentriamTable.js]: :+1:

**Review by karlrwjohnson — APPROVED** [review]

**Pull request merged**